src/network_protocol/data_link.rs

```diff
@@ -62,7 +63,11 @@ impl<'a> DataLink<'a> {
     //    const BVLC_ORIGINAL_BROADCAST_NPDU: u8 = 11;
 
     pub fn new(function: DataLinkFunction, npdu: Option<NetworkPdu<'a>>) -> Self {
-        Self { function, npdu }
+        Self {
+            function,
+            npdu,
+            raw_payload: &[],
```

Comment:
As mentioned in a comment in this PR, this should not be here as keeping track of segments is not the job of the codec. It should be done at a higher level.